packages/d3fc-series/src/webgl/boxPlot.js

```diff
@@ -0,0 +1,78 @@
+import boxPlotBase from '../boxPlotBase';
+import { glBoxPlot, scaleMapper } from '@d3fc/d3fc-webgl';
+import { rebindAll, exclude, rebind } from '@d3fc/d3fc-rebind';
+import functor from '../functor';
+
+export default () => {
+    const base = boxPlotBase();
+    let cap = functor(0.5);
+
+    let draw = glBoxPlot();
+
+    const boxPlot = (data) => {
+        const filteredData = data.filter(base.defined());
+
+        const xScale = scaleMapper(base.xScale());
+        const yScale = scaleMapper(base.yScale());
+
+        const accessor = getAccessors();
+        
+        const xValues = new Float32Array(filteredData.length);
+        const medianValues = new Float32Array(filteredData.length);
+        const upperQuartileValues = new Float32Array(filteredData.length);
+        const lowerQuartileValues = new Float32Array(filteredData.length);
+        const highValues = new Float32Array(filteredData.length);
+        const lowValues = new Float32Array(filteredData.length);
+        const bandwidth = new Float32Array(filteredData.length);
+        const capWidth = new Float32Array(filteredData.length);
+
+        filteredData.forEach((d, i) => {
+            xValues[i] = xScale.scale(accessor.xValues(d, i));
+            medianValues[i] = yScale.scale(accessor.medianValues(d, i));
+            upperQuartileValues[i] = yScale.scale(accessor.upperQuartileValues(d, i));
+            lowerQuartileValues[i] = xScale.scale(accessor.lowerQuartileValues(d, i));
+            highValues[i] = yScale.scale(accessor.highValues(d, i));
+            lowValues[i] = yScale.scale(accessor.lowValues(d, i));
+            bandwidth[i] = accessor.bandwidth(d, i);
+            capWidth[i] = bandwidth[i] * cap(d, i);
+        });
+
+        draw.xValues(xValues)
+            .medianValues(medianValues)
+            .upperQuartileValues(upperQuartileValues)
+            .lowerQuartileValues(lowerQuartileValues)
+            .highValues(highValues)
+            .lowValues(lowValues)
+            .bandwidth(bandwidth)
+            .capWidth(capWidth)
+            .xScale(xScale.glScale)
+            .yScale(yScale.glScale);
+
+        draw(filteredData.length);
+    };
+
+    function getAccessors() {
```

Comment:
Any reason for this being pulled out into a separate function rather than the values being inlined?

Reply:
Mostly it's for consistency - we've done the same thing in errorBar, line, ohlc and point. It was originally because line and point needed a way to handle orient. If you want I can remove it from errorBar, ohlc and boxPlot?